### PR TITLE
remove redux-modules

### LINKED ISSUE #1148: Consider to get rid of redux-modules in order to decrease the size of the bundle file
redux-modules has ramda dependency and it adds up 273 Kb alone to the output bundle

### PR DESCRIPTION
**Pull request by bnchdrff**
closes #1148 

see #1797 and #2020 for previous work & discussion

abandon redux-modules and switch to ducks-style organization of redux code

this PR excludes a switch to redux-immutable -- it causes problems with our redux-form version - i tried updating to 7.x but it'd be a lot of work.

### CODE REVIEW

File: src/app/components/App.jsx [hunk 1]
```diff
@@ -331,12 +330,11 @@ export default connect(
     },
     dispatch => ({
         loginUser: () =>
-            dispatch(user.actions.usernamePasswordLogin()),
+            dispatch(userActions.usernamePasswordLogin({})),
```
Comment (relativityboy) [inline src/app/components/App.jsx]: The change from user.actions* to userActions was to preserve `redux/User` imports?

File: src/app/components/elements/ChangePassword.jsx [hunk 2]
```diff
@@ -262,11 +264,11 @@ export default reduxForm(
             }))
         },
         notify: (message) => {
-            dispatch({type: 'ADD_NOTIFICATION', payload: {
+            dispatch(appActions.addNotification({
                 key: "chpwd_" + Date.now(),
                 message,
-                dismissAfter: 5000}
-            });
+                dismissAfter: 5000,
```
Comment (relativityboy) [inline src/app/components/elements/ChangePassword.jsx]: gonna take a while to get used to dangling commas... are those allowed by our style-guide?

File: src/app/components/cards/CardView.js [hunk 3]
```diff
@@ -1,7 +1,7 @@
 import React from 'react';
 import {connect} from 'react-redux'
 import Link from 'app/components/elements/Link'
-import g from 'app/redux/GlobalReducer'
+import * as globalActions from 'app/redux/GlobalReducer';
```
Comment (relativityboy) [inline src/app/components/cards/CardView.js]: Thank you.

File: src/app/components/elements/MarkNotificationRead.jsx [hunk 4]
```diff
@@ -27,5 +27,5 @@ class MarkNotificationRead extends React.Component {
 }
 
 export default connect(null, dispatch => ({
-    update: (payload) => { dispatch({type: 'UPDATE_NOTIFICOUNTERS', payload})},
+    update: (payload) => { dispatch({type: 'app/UPDATE_NOTIFICOUNTERS', payload})},
```
Comment (relativityboy) [inline src/app/components/elements/MarkNotificationRead.jsx]: Does this deserve an action creator? à la `appActions.addNotification` ?

File: src/app/redux/AppReducer.js [hunk 5]
```diff
@@ -1,13 +1,23 @@
...
+
+const STEEM_API_ERROR = 'app/STEEM_API_ERROR';
+const FETCH_DATA_BEGIN = 'app/FETCH_DATA_BEGIN';
+const FETCH_DATA_END = 'app/FETCH_DATA_END';
+const ADD_NOTIFICATION = 'app/ADD_NOTIFICATION';
+const REMOVE_NOTIFICATION = 'app/REMOVE_NOTIFICATION';
+const UPDATE_NOTIFICOUNTERS = 'app/UPDATE_NOTIFICOUNTERS';
+const SET_USER_PREFERENCES = 'app/SET_USER_PREFERENCES';
+const TOGGLE_NIGHTMODE = 'app/TOGGLE_NIGHTMODE';
+const TOGGLE_BLOGMODE = 'app/TOGGLE_BLOGMODE';
```
Comment (relativityboy) [inline src/app/redux/AppReducer.js]: 👍

File: src/app/redux/AppReducer.js [hunk 6]
```diff
@@ -24,55 +34,93 @@ const defaultState = Map({
...
+            return state.set('user_preferences', Map(action.payload));
+        case TOGGLE_NIGHTMODE:
+            return state.setIn(['user_preferences', 'nightmode'], !state.getIn(['user_preferences', 'nightmode']));
+        case TOGGLE_BLOGMODE:
+            return state.setIn(['user_preferences', 'blogmode'], !state.getIn(['user_preferences', 'blogmode']));
+        default:
+            return state;
     }
-    return res;
 }
```
Comment (relativityboy) [inline src/app/redux/AppReducer.js]: The `notifications-ui` branch does exactly this as well. You probably know that. We'll need to keep tabs when merging it in.

File: src/app/redux/SagaShared.js [hunk 7]
```diff
@@ -85,5 +88,5 @@ function* saveUserPreferences({payload}) {
 }
 
 function* watchUserSettingsUpdates() {
-    yield* takeLatest(['SET_USER_PREFERENCES', 'TOGGLE_NIGHTMODE', 'TOGGLE_BLOGMODE'], saveUserPreferences);
+    yield* takeLatest(['app/SET_USER_PREFERENCES', 'app/TOGGLE_NIGHTMODE', 'app/TOGGLE_BLOGMODE'], saveUserPreferences);
```
Comment (relativityboy) [inline src/app/redux/SagaShared.js]: Wondering if these strings should be constants.

File: src/app/redux/tests/global.test.js [hunk 8]
```diff
@@ -3,11 +3,9 @@
 import chai, {expect} from 'chai';
 import chaiImmutable from 'chai-immutable';
 import Immutable, {Map} from 'immutable';
-import ReducerModule from '../GlobalReducer';
+import reducer, * as actions from '../GlobalReducer';
```
Comment (relativityboy) [inline src/app/redux/tests/global.test.js]: `* as globalActions` would be better, for consistency.

File: src/app/components/App.jsx (hunk 1 above)
Comment (bnchdrff) [reply src/app/components/App.jsx]: yes -- redux-modules enforced the actions child object

File: src/app/components/elements/ChangePassword.jsx (hunk 2 above)
Comment (bnchdrff) [reply src/app/components/elements/ChangePassword.jsx]: they're not disallowed: https://github.com/steemit/condenser/blob/master/.eslintrc#L17

Comment (bnchdrff) [reply src/app/components/elements/ChangePassword.jsx]: and airbnb/javascript wants them https://github.com/airbnb/javascript#commas--dangling

we should enforce dangling commas; they make diffs way easier to read.

File: src/app/components/elements/MarkNotificationRead.jsx (hunk 4 above)
Comment (bnchdrff) [reply src/app/components/elements/MarkNotificationRead.jsx]: oops it actually has one and i forgot to change this guy

File: src/app/redux/AppReducer.js (hunk 6 above)
Comment (bnchdrff) [reply src/app/redux/AppReducer.js]: not in this reducer though -- we're only working in new reducers so it'll be an easy merge

File: src/app/redux/SagaShared.js (hunk 7 above)
Comment (bnchdrff) [reply src/app/redux/SagaShared.js]: we could export the action types & make them available for the sagas. also it'd be good to explicitly note in the reducers when an action has a saga, like has been done here https://github.com/steemit/condenser/blob/master/src/app/redux/User.js#L92

i'd rather make these changes in a future PR though

File: src/app/redux/tests/global.test.js (hunk 8 above)
Comment (bnchdrff) [reply src/app/redux/tests/global.test.js]: eh it's a test

File: src/app/redux/SagaShared.js (hunk 7 above)
Comment (bnchdrff) [reply src/app/redux/SagaShared.js]: actually i'm doing it now

File: src/app/redux/GlobalReducer.js [hunk 9]
```diff
@@ -1,302 +1,424 @@
...
+const SHOW_DIALOG = 'global/SHOW_DIALOG';
+const HIDE_DIALOG = 'global/HIDE_DIALOG';
+// Saga-related:
+export const GET_STATE = 'global/GET_STATE';
+
+export default function reducer(state = defaultState, action) {
+    const payload = action.payload;
+
+    switch (action.type) {
+        case SET_COLLAPSED: {
```
Comment (relativityboy) [inline src/app/redux/GlobalReducer.js]: Using a switch block like this is koo (if/else). I'd suggest alphabetizing the `case:` params. It'll make it easier to use, and might actually make it easier to review. (unless there are perf-reasons or non- `return;' / 'break;` blocks.

File: src/app/redux/TransactionReducer.js [hunk 10]
```diff
@@ -0,0 +1,187 @@
...
+                        } else if (errorStr === 'Testing, fake error')
+                            errorKey = 'Testing, fake error';
+                        break;
+                    case 'transfer':
+                        if (/get_balance/.test(errorStr)) {
+                            errorKey = 'Insufficient balance.'
+                        }
+                        break;
+                    case 'withdraw_vesting':
+                        if(/Account registered by another account requires 10x account creation fee worth of Steem Power/.test(errorStr))
```
Comment (relativityboy) [inline src/app/redux/TransactionReducer.js]: This regex isn't going to change. It should be a const with a nice descriptive name.

File: src/app/redux/TransactionReducer.js [hunk 11]
```diff
@@ -0,0 +1,187 @@
...
+            // See TransactionSaga.js
+            return state;
+
+        case 'transaction/UPDATE_AUTHORITIES':
+            return state;
+
+        case 'transaction/UPDATE_META':
+            return state;
+
+        case 'transaction/ERROR': {
```
Comment (relativityboy) [inline src/app/redux/TransactionReducer.js]: The regexes in this block aren't going to change. They should be consts with nice descriptive names.

File: src/app/redux/GlobalReducer.js (hunk 9 above)
Comment (bnchdrff) [reply src/app/redux/GlobalReducer.js]: i kept the original order. i don't want to get too deep into the weeds of totally rewriting the reducers at this point; open an issue for this one

File: src/app/redux/TransactionReducer.js (hunk 11 above)
Comment (bnchdrff) [reply src/app/redux/TransactionReducer.js]: mind opening an issue for this?

Comment (bnchdrff) [reply src/app/redux/TransactionReducer.js]: actually it's covered already by https://github.com/steemit/condenser/issues/2037 possibly -- leave a comment pointing to this code

File: src/app/redux/UserReducer.js [hunk 12]
```diff
@@ -0,0 +1,345 @@
...
+
+if (process.env.BROWSER) {
+    const locale = store.get('language');
+    if (locale) defaultState.locale = locale;
+}
+
+export default function reducer(state = defaultState, action) {
+    const payload = action.payload;
+
+    switch (action.type) {
```
Comment (relativityboy) [inline src/app/redux/UserReducer.js]: Alphabetize cases.

File: src/app/redux/TransactionReducer.js [hunk 13]
```diff
@@ -0,0 +1,187 @@
...
+const defaultState = fromJS({
+    operations: [],
+    status: { key: '', error: false, busy: false, },
+    errors: null
+});
+
+export default function reducer(state = defaultState, action) {
+    const payload = action.payload;
+
+    switch (action.type) {
```
Comment (relativityboy) [inline src/app/redux/TransactionReducer.js]: Should these case statements be constants? Should they be exportable? 
Alphabetize.

File: src/app/redux/UserReducer.js (hunk 12 above)
Comment (bnchdrff) [reply src/app/redux/UserReducer.js]: i really don't think alphabetical order makes sense actually. for example, CONFIRM_OPERATION and HIDE_CONFIRM should be adjacent.

File: src/app/redux/TransactionReducer.js (hunk 13 above)
Comment (bnchdrff) [reply src/app/redux/TransactionReducer.js]: oops yes they should be constants.. changing this now.

File: src/app/redux/TransactionReducer.js (hunk 11 above)
Comment (relativityboy) [reply src/app/redux/TransactionReducer.js]: Added, though I think it'd make sense to create the consts now. Let the testing be focused on tests.

File: src/app/redux/tests/global.test.js (hunk 8 above)
Comment (relativityboy) [reply src/app/redux/tests/global.test.js]: That's just sloppy.

File: src/app/redux/UserReducer.js (hunk 12 above)
Comment (relativityboy) [reply src/app/redux/UserReducer.js]: That suggests inconsistent naming choices. Alphabetization is 'a' stricture that helps us uncover poor choices like that.

Comment (bnchdrff) [reply src/app/redux/UserReducer.js]: I'm not going to rename all of the action types in this branch. if you think this is important you can make an issue and the team can discuss.

File: src/app/redux/GlobalReducer.js (hunk 9 above)
Comment (relativityboy) [reply src/app/redux/GlobalReducer.js]: Added #2072

File: src/app/redux/UserReducer.js (hunk 12 above)
Comment (relativityboy) [reply src/app/redux/UserReducer.js]: We'll fold that, and the reordering into #2072

**Comment by bnchdrff** [comment]
alright @valzav @relativityboy @gl2748 this branch is now deployed at https://testnet.steemitdev.com -- please give it a nice round of posting/upvoting/etc and watch the console to make sure i didn't fuck something up

**Pull request merged**